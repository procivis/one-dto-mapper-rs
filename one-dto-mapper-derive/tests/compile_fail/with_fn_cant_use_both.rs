use one_dto_mapper_derive::From;

struct NumericDto {
    age: u16,
}

#[derive(From)]
#[from(NumericDto)]
struct StringDto {
    #[from(
        with_fn = custom_to_string,
        with_fn_ref = custom_to_string_from_ref
    )]
    age: String,
}

fn custom_to_string(value: u16) -> String {
    value.to_string()
}

fn custom_to_string_from_ref(value: &u16) -> String {
    value.to_string()
}

fn main() {}
