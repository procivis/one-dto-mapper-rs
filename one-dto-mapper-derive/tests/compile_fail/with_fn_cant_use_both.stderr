error: Only one can be used: (with_fn, with_fn_ref) at age
 --> tests/compile_fail/with_fn_cant_use_both.rs:7:10
  |
7 | #[derive(From)]
  |          ^^^^
  |
  = note: this error originates in the derive macro `From` (in Nightly builds, run with -Z macro-backtrace for more info)
